bamboo/unit_tests/test_unit_checkpoint.py

```diff
@@ -22,7 +22,7 @@ def skeleton_checkpoint_lenet_shared(cluster, executables, dir_name,
         cluster=cluster, executable=exe, num_nodes=1, num_processes=2,
         dir_name=dir_name,
         data_filedir_default='/p/lscratchh/brainusr/datasets/MNIST',
-        data_reader_name='mnist', data_reader_percent=data_reader_percent,
+        data_reader_name='mnist', data_reader_percent=0.01,
```

Comment:
Are we always doing this at 1% even for weekly?

Reply:
Same question as Brian.

Possible alternatives:
-Have Nightly run at 10% of passed-in value and Weekly at 100% (requires parsing prototext to get `data_reader_percent`)
-Rework the logic in `tools.get_command` to run 100% for Weekly no matter what and then use the prototext value for Nightly.

Reply:
I don't think this test in particular should ever run at any value other than "1 sample" -- this test should be data-invariant; as such, nothing new is tested by running 1 billion samples vs a single sample. As discussed with @bvanessen, I will update this to allow a value to be taken in but if `data_reader_percent` is `None`, then it will hard-code `0.01`. Does this seem reasonable?

Reply:
@benson31 yes that seems reasonable.  @forsyth2 If a prototext specifies a data reader percentage, then I do not think that the test should override it.  Specifically, I can imagine many future tests that require a precise number of samples to check for weird corner cases.

Reply:
@bvanessen @forsyth2 The caveat to that is that the data readers in `model_zoo/` might specify `data_reader_percent` for any number of reasons (e.g., the data set is huge and someone put a reasonable value for debugging and then aggressively committed). The testing infrastructure should be completely isolated from the rest of the world. I think that if a test requires a specific value, it should do one of the following, in decreasing order of desirability:
- Specify the value explicitly in PyTest.
- Copy the data reader file to a testing-specific location and use that with the `data_reader_percent='prototext'` option.
Moreover, if a test requires a specific amount of data to pass, users should not be able to command-line override that amount for those tests (because then the test will fail for a non-code reason (in particular, the contract of the test will have been broken)).